internal/controller/pullrequest_controller.go

```diff
@@ -60,119 +60,79 @@ func (r *PullRequestReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	logger := log.FromContext(ctx)
 
 	var pr promoterv1alpha1.PullRequest
-	err := r.Get(ctx, req.NamespacedName, &pr, &client.GetOptions{})
-	if err != nil {
+	if err := r.Get(ctx, req.NamespacedName, &pr); err != nil {
 		if errors.IsNotFound(err) {
 			logger.Info("PullRequest not found", "namespace", req.Namespace, "name", req.Name)
 			return ctrl.Result{}, nil
 		}
-
-		logger.Error(err, "failed to get PullRequest", "namespace", req.Namespace, "name", req.Name)
 		return ctrl.Result{}, fmt.Errorf("failed to get PullRequest: %w", err)
 	}
 
-	pullRequestProvider, err := r.getPullRequestProvider(ctx, pr)
-	if err != nil {
+	provider, err := r.getPullRequestProvider(ctx, pr)
+	if err != nil || provider == nil {
```

Comment:
Instead of the double check, just return an error from `getPullRequestProvider`?